### PR TITLE
[MM-57721] Convert `./components/channel_header_dropdown/menu_items/leave_channel/leave_channel.tsx` from Class Component to Function Component

### PR DESCRIPTION
**Pull request by Syed-Ali-Abbas-Zaidi**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/26689
Jira https://mattermost.atlassian.net/browse/MM-57721

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/channel_header_dropdown/menu_items/leave_channel/leave_channel.tsx [hunk 1]
```diff
@@ -64,19 +62,17 @@ export default class LeaveChannel extends React.PureComponent<Props> {
...
-    }
-}
+    }, [channel, leaveChannel, openModal]);
+
+    return (
+        <Menu.ItemAction
+            id={id}
+            show={(!isDefault || isGuestUser) && channel.type !== Constants.DM_CHANNEL && channel.type !== Constants.GM_CHANNEL}
+            onClick={handleLeave}
+            text={intl.formatMessage({id: 'channel_header.leave', defaultMessage: 'Leave Channel'})}
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/leave_channel/leave_channel.tsx:72]: :heart:

**Review by larkox — APPROVED** [review]
Great work! Thanks!

**Review by devinbinnie — APPROVED** [review]
LGTM! Thanks @Syed-Ali-Abbas-Zaidi!

**Comment by devinbinnie** [comment]
@Syed-Ali-Abbas-Zaidi looks like the type checks and tests are failing. Can you have a look at those?

Also @larkox do these tickets need QA review?

**Comment by larkox** [comment]
@devinbinnie Re: QA
Better safe than sorry :smile:

**Review by yasserfaraazkhan — APPROVED** [review]
🎉

**Pull request merged**